plugins/boltz/main.go

```diff
@@ -3,15 +3,63 @@ package boltz
 import (
 	"time"
 
+	"github.com/BoltzExchange/boltz-lnd/boltz"
+	"github.com/bolt-observer/agent/entities"
 	agent_entities "github.com/bolt-observer/agent/entities"
+	"github.com/btcsuite/btcd/chaincfg"
 	"github.com/golang/glog"
+	"github.com/urfave/cli"
 )
 
+const (
+	DefaultBoltzUrl = "https://boltz.exchange/api"
+)
+
+var PluginFlags = []cli.Flag{
+	cli.StringFlag{
+		Name: "boltzurl", Value: DefaultBoltzUrl, Usage: "url of boltz api", Hidden: false,
+	},
+}
+
 // Plugin can save its data here
 type Plugin struct {
+	BoltzAPI    *boltz.Boltz
+	ChainParams *chaincfg.Params
+	LnAPI       entities.NewAPICall
+
 	agent_entities.Plugin
 }
 
+// NewBoltzPlugin creates new instance
+func NewBoltzPlugin(lnAPI agent_entities.NewAPICall, cmdCtx *cli.Context) *Plugin {
+	network := cmdCtx.String("network")
+
+	params := chaincfg.MainNetParams
+	switch network {
+	case "mainnet":
+		params = chaincfg.MainNetParams
+	case "testnet":
+		params = chaincfg.TestNet3Params
+	case "regtest":
+		params = chaincfg.RegressionNetParams
+	case "simnet":
+		params = chaincfg.SimNetParams
+	}
+
+	resp := &Plugin{
+		ChainParams: &params,
+		BoltzAPI: &boltz.Boltz{
+			URL: cmdCtx.String("boltzurl"),
+		},
+	}
+	if lnAPI == nil {
```

Comment:
Do we really need this check?

Reply:
Refactored everything so err is reported there (yeah know I still need to remove this nil check)